src/vcpkg/base/system.mac.cpp

```diff
@@ -0,0 +1,262 @@
+
+        // ignore "network adapter"
+        if (parser.require_character('"')) return false;
+        parser.match_until(is_quote);
+        if (parser.require_character('"')) return false;
+        if (parser.require_character(',')) return false;
+
+        // get "physical address"
+        if (parser.require_character('"')) return false;
+        auto mac_address = parser.match_until(is_quote).to_string();
+        if (parser.require_character('"')) return false;
+        if (parser.require_character(',')) return false;
+
+        // ignore "transport name"
+        if (parser.require_character('"')) return false;
+        parser.match_until(is_quote);
+        if (parser.require_character('"')) return false;
+
+        if (!parser.at_eof()) return false;
+
+        // output line was properly formatted
+        out = Strings::ascii_to_lowercase(Strings::replace_all(mac_address, "-", ":"));
+        return true;
+    }
+
+    std::string get_user_mac_hash()
+    {
+#if defined(_WIN32)
+        // getmac /V /NH /FO CSV
+        // outputs each interface on its own comma-separated line
+        // "connection name","network adapter","physical address","transport name"
+        auto getmac = cmd_execute_and_capture_output(
+            Command("getmac").string_arg("/V").string_arg("/NH").string_arg("/FO").string_arg("CSV"));
+        if (getmac.exit_code == 0)
+        {
+            for (auto&& line : Strings::split(getmac.output, '\n'))
+            {
+                std::string mac;
+                if (extract_mac_from_getmac_output_line(line, mac) && is_valid_mac_for_telemetry(mac))
+                {
+                    return Hash::get_string_hash(mac, Hash::Algorithm::Sha256);
+                }
+            }
+        }
+#elif defined(__linux__) || defined(__APPLE__)
+        // The getifaddrs(ifaddrs** ifap) function creates a linked list of structures
+        // describing the network interfaces of the local system, and stores
+        // the address of the first item of the list in *ifap.
+        // man page: https://www.man7.org/linux/man-pages/man3/getifaddrs.3.html
+        struct ifaddrs_guard
+        {
+            ifaddrs* ptr = nullptr;
+            ~ifaddrs_guard()
+            {
+                if (ptr) freeifaddrs(ptr);
+            }
+        } interfaces;
+
+        if (getifaddrs(&interfaces.ptr) < 0)
+        {
+            return "0";
+        }
+
+        unsigned char bytes[MAC_BYTES_LENGTH];
+        for (auto interface = interfaces.ptr; interface; interface = interface->ifa_next)
+        {
+            // The ifa_addr field points to a structure containing the interface
+            // address.  (The sa_family subfield should be consulted to
+            // determine the format of the address structure.)  This field may
+            // contain a null pointer.
+            if (interface->ifa_addr && interface->ifa_addr->sa_family == AF_TYPE)
+            {
```

Comment:
```suggestion
            if (!interface->ifa_addr || interface->ifa_addr->sa_family != AF_TYPE)
            {
            continue;
            }
```

(and then deindent below and remove })